### ISSUE TITLE
solc 0.3.6 is breaking behaviour

### ISSUE THREAD

**Issue by mhhf** [nexusdev/dapple#299]
```
dapple test
Testing...
You are on a MORDEN chain. Forking of might take a while...
<stdin>:2:1: Warning: Source file does not specify required compiler version! Consider adding "pragma solidity ^0.3.6;".

^

A
  Error: Contract endowment error: Error: VM Exception while executing transaction: invalid JUMP
/Users/mhhf/work/99_extern/nexus/dapple/dapple/node_modules/solc/soljson.js:1
(function (exports, require, module, __filename, __dirname) { var Module;if(!Module)Module=(typeof Module!=="undefined"?Module:null)||{};var moduleOverrides={};for(var key in Module){if(Module.hasOwnProperty(key)){moduleOverrides[key]=Module[key]}}var ENVIRONMENT_IS_WEB=typeof window==="object";var ENVIRONMENT_IS_WORKER=typeof importScripts==="function";var ENVIRONMENT_IS_NODE=typeof process==="object"&&typeof require==="function"&&!ENVIRONMENT_IS_WEB&&!ENVIRONMENT_IS_WORKER;var ENVIRONMENT_IS_SHELL=!ENVIRONMENT_IS_WEB&&!ENVIRONMENT_IS_NODE&&!ENVIRONMENT_IS_WORKER;if(ENVIRONMENT_IS_NODE){if(!Module["print"])Module["print"]=function print(x){process["stdout"].write(x+"\n")};if(!Module["printErr"])Module["printErr"]=function printErr(x){process["stderr"].write(x+"\n")};var nodeFS=require("fs");var nodePath=require("path");Module["read"]=function read(filename,binary){filename=nodePath["normalize"](filename);var ret=nodeFS["readFileS

TypeError: Cannot read property 'indexOf' of undefined
    at Object.transformToFullName (/Users/mhhf/work/99_extern/nexus/dapple/dapple-core/node_modules/web3/lib/utils/utils.js:192:18)
    at Contract.abi.forEach.abi (/Users/mhhf/work/99_extern/nexus/dapple/dapple-core/contract.js:46:80)
    at Array.forEach (native)
    at new Contract (/Users/mhhf/work/99_extern/nexus/dapple/dapple-core/contract.js:42:6)
    at DestroyableTransform._transform (/Users/mhhf/work/99_extern/nexus/dapple/dapple-test/lib/test.js:133:24)
    at DestroyableTransform.Transform._read (/Users/mhhf/work/99_extern/nexus/dapple/dapple-test/node_modules/through2/node_modules/readable-stream/lib/_stream_transform.js:159:10)
    at DestroyableTransform.Transform._write (/Users/mhhf/work/99_extern/nexus/dapple/dapple-test/node_modules/through2/node_modules/readable-stream/lib/_stream_transform.js:147:83)
    at doWrite (/Users/mhhf/work/99_extern/nexus/dapple/dapple-test/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:313:64)
    at writeOrBuffer (/Users/mhhf/work/99_extern/nexus/dapple/dapple-test/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:302:5)
    at DestroyableTransform.Writable.write (/Users/mhhf/work/99_extern/nexus/dapple/dapple-test/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:241:11)
```
There seems to be something wrong with the test contract as it suddenly throws

**Comment by nmushegian** [comment]
> `Warning: Source file does not specify required compiler version! Consider adding "pragma solidity ^0.3.6;".`

Yep. Anything without a pragma tag is legacy. Solution is to update sources.

There's a few other updates we have to do with every package like turning local imports into "./foo.sol" form

**Comment by mhhf** [comment]
pragma tag is a warning and don't cause this error. The ABI descriptor has changed significantly and i need to update dapples code in various places where we work with solidities abi output. I'm on it currently, should be solved by today.

**Comment by mhhf** [comment]
Error was because from solidity 0.3.6 upwards, contract's fallback function throws by default, if not implemented but used. with this you have to explicitly implement an empty fallback function in case you want to be able to send funds to a contract. For dapple-test we had "fueled" a contract after deployment without having a fallback function, this caused an error during test contract setUp. Its fixed now in 0.8
